"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorBoundary = void 0;
var react_1 = require("react");
var components_1 = require("@storybook/components");
var theming_1 = require("@storybook/theming");
var ErrorBoundary = (function (_super) {
    __extends(ErrorBoundary, _super);
    function ErrorBoundary() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = { hasError: false };
        return _this;
    }
    ErrorBoundary.getDerivedStateFromError = function (error) {
        return {
            hasError: true,
            error: error,
        };
    };
    ErrorBoundary.prototype.componentDidCatch = function (error, info) {
        console.group("An error occurred during rendering Addon panel of storybook-addon-designs");
        console.log("--- Error ---");
        console.error(error);
        console.log("--- React Component Stack ---");
        console.error(info.componentStack);
        console.groupEnd();
    };
    ErrorBoundary.prototype.render = function () {
        if (this.state.hasError) {
            return ((0, theming_1.jsx)(components_1.Placeholder, null,
                (0, theming_1.jsx)(react_1.Fragment, null, "Failed to render addon UI"),
                (0, theming_1.jsx)(react_1.Fragment, null,
                    (0, theming_1.jsx)("p", null, "Sorry, this addon has crashed due to the below error has thrown during rendering the addon UI."),
                    (0, theming_1.jsx)("pre", null, String(this.state.error)),
                    (0, theming_1.jsx)("p", null,
                        "See console log for more details. To clear the error state, please reload the page.",
                        " ",
                        (0, theming_1.jsx)(components_1.Link, { href: "https://github.com/storybookjs/addon-designs/issues/new?assignees=&labels=category%3A+bug&template=bug_report.yml", target: "_blank", rel: "noopener", withArrow: true, cancel: false }, "Bug report")))));
        }
        return this.props.children;
    };
    return ErrorBoundary;
}(react_1.Component));
exports.ErrorBoundary = ErrorBoundary;
